src/sdk/namespace_fs.js

```diff
@@ -601,6 +603,7 @@ class NamespaceFS {
         let source_file;
         let buffer_pool_cleanup = null;
         try {
+            await this._fail_if_archived_or_sparse_file(fs_account_config, source_file_path);
```

Comment:
I think this is not the right spot - we said to check and fail even if we do link.
So this should be called in the upper level in `upload_object()` under the `if (params.copy_source) { ...` block